WorkspaceServer/WorkspaceFeatures/PackageTool.cs

```diff
@@ -30,22 +30,39 @@ public PackageTool(string name, IDirectoryAccessor directoryAccessor)
 
         public async Task<ProjectAsset> LocateProjectAsset()
         {
-            var result = await CommandLine.Execute(_path.Value, MLS.PackageTool.PackageToolConstants.LocateProjectAsset, DirectoryAccessor.GetFullyQualifiedRoot());
-            var projectDirectory = new DirectoryInfo(string.Join("", result.Output));
-            return new ProjectAsset(new FileSystemDirectoryAccessor(projectDirectory));
+            if (Exists)
+            {
+                var result = await CommandLine.Execute(_path.Value, MLS.PackageTool.PackageToolConstants.LocateProjectAsset, DirectoryAccessor.GetFullyQualifiedRoot());
+                if (result.ExitCode == 0)
+                {
+                    var projectDirectory = new DirectoryInfo(string.Join("", result.Output));
```

Comment:
Given the failure modes I've seen in the past it would be nice if this checked for an empty string, otherwise internally `Path.GetFullPath()` will throw and it's not really helpful.  E.g.,

``` c#
var path = string.Join("", result.Output);
if (string.IsNullOrEmpty(path))
{
    // throw, return null, something else
}
var projectDirectory = new DirectoryInfo(path);
```